components/sidebar.jsx

```diff
@@ -627,12 +633,36 @@ export default class Sidebar extends React.Component {
 
         const displayName = channel.display_name;
 
+        const channelLink = this.createChannelButtonOrLink(link, rowClass, icon, displayName, badge, closeButton);
+
         return (
             <li
                 key={channel.name}
                 ref={channel.name}
                 className={linkClass}
             >
+                {channelLink}
+                {tutorialTip}
+            </li>
+        );
+    }
+
+    createChannelButtonOrLink(link, rowClass, icon, displayName, badge, closeButton) {
+        let element;
+        if (isDesktopApp()) {
+            element = (
+                <button
+                    className={'btn btn-link ' + rowClass}
+                    onClick={() => this.handleClick(link)}
```

Comment:
No change needed right now since this component isn't particularly efficient, but you'd usually want to avoid using an arrow function in component prop since that'll mean it's making a new function on every re-render. That's not that bad when it's a DOM element like button, but it'll cause PureComponents to re-render more often than necessary